tests/eas/cpuhp.py

```diff
@@ -0,0 +1,217 @@
+        sleep_min = cls.hp_stress['sleep']['min_ms']
+        sleep_max = cls.hp_stress['sleep']['max_ms']
+        with open(script_path, 'w') as script:
+            if cls.target.os == 'android':
+                script.write('#!/system/bin/sh\n')
+            else:
+                script.write('#!/bin/sh\n')
+            cls.pid_file = cls.target.working_directory + '/hotplug.pid'
+            script.write('echo $$ > {}\n'.format(cls.pid_file))
+            script.write('while true; do\n')
+            for cpu, plug_way in cls._random_cpuhp_seq():
+                # Write in sysfs entry
+                sysfs_entry = hp._cpu_path(cls.target, cpu)
+                cmd = 'echo {} > {}\n'.format(plug_way, sysfs_entry)
+                script.write(shift + cmd)
+                # Log in trace file
+                msg = 'Plug {} CPU {}'.format('in' if plug_way else 'off', cpu)
+                script.write(shift + 'echo "{}" > {}\n'.format(msg, marker))
+                # Sleep if necessary
+                if sleep_max > 0:
+                    sleep_dur_sec = random.randint(sleep_min, sleep_max)/1000.0
+                    script.write(shift + 'sleep {}\n'.format(sleep_dur_sec))
+            script.write('done\n') # end of while
+        return script_path
+
+    @classmethod
+    def _random_cpuhp_seq(cls):
+        '''
+        Yield a consitent random sequence of CPU hotplug operations
+        
+        "Consistent" means that a CPU will be plugged-in only if it was 
+        plugged-off before (and vice versa). Moreover the state of the CPUs
+        once the sequence has completed should the same as it was before. The
+        length of the sequence (that is the number of plug operations) is 
+        defined in the hp_stress member of the class. The actual length of the
+        sequence might differ from the requested one by 1 because it's easier 
+        to implement and it shouldn't be an issue for most test cases.
+        '''
+        cur_on_cpus = cls.hotpluggable_cpus[:]
+        cur_off_cpus = []
+        max_cpus_off = cls.hp_stress['max_cpus_off']
+        i = 0
+        while i < cls.hp_stress['sequence_len'] - len(cur_off_cpus):
+            if len(cur_on_cpus)<=1 or len(cur_off_cpus)>=max_cpus_off:
+                # Force plug IN when only 1 CPU is on or too many are off
+                plug_way = 1
+            elif not cur_off_cpus:
+                # Force plug OFF if all CPUs are on
+                plug_way = 0 # Plug OFF
+            else:
+                plug_way = random.randint(0,1)
+            src = cur_off_cpus if plug_way else cur_on_cpus
+            dst = cur_on_cpus if plug_way else cur_off_cpus
+            cpu = random.choice(src)
+            src.remove(cpu)
+            dst.append(cpu)
+            i += 1
+            yield cpu, plug_way
+
+        # Re-plug offline cpus to come back to original state
+        for cpu in cur_off_cpus:
+            yield cpu, 1
```

Comment:
And actually, the sequence will need to be long enough to cover _all iterations_ of the test, otherwise the last ones might run without hotplug pressure. It's maybe just simpler to leave it in a loop :)